erpc/erpc.go

```diff
@@ -64,11 +63,11 @@ type Client struct {
 func (c *Client) NewRequest(request []byte) []byte {
```

Comment:
We shouldn't worry about it now, but at some point, I think it would be cool to not need to copy the args.
E.g. find a way to avoid marshaling the request and then copying that byte slice into another byte slice. That would require that the slice with the cid,seq is allocated first, then the user of erpc puts their args into that byte slice.

Alternatively, we could put the cid,seq at the end of the byte slice. Then, we can just append 16 bytes to the end of the user's slice, and there's a chance that it won't need to be copied (or e.g. the user of erpc could just make sure to put 16 extra bytes in the capacity before passing `request` into `NewRequest`).

I don't think this is a big deal, because I don't have any evidence that this is a major source of performance limitations.

Reply:
Yeah, I think we should change the semantics of a 'handler' to be to *append* its reply to the existing buffer, rather than overwriting the buffer.

But we still won't be able to pre-allocate a buffer that is definitely big enough, so there might be some copy overhead there.